examples/client_batch.ml

```diff
@@ -0,0 +1,47 @@
+(*
+ * Copyright (c) 2018-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open Lwt.Syntax
+open Lwt.Infix
+module Store = Irmin_mem.KV.Make (Irmin.Contents.String)
+module Client = Irmin_client_unix.Make (Store)
+module Error = Irmin_client.Error
+
+let main =
+  let info () = Client.Info.empty in
+  let uri = Uri.of_string Sys.argv.(1) in
+  let* client = Client.connect uri in
+
+  let* main = Client.main client in
+  let* () = Client.set_exn ~info main [ "testing" ] "testing" in
+  let* head = Client.Branch.get client Client.Branch.main in
+  let* tree =
+    Client.Batch.Tree.of_commit client (Client.Commit.hash head) >|= Option.get
+  in
+  let* tree = Client.Batch.Tree.add client tree [ "b"; "c" ] "123" in
```

Comment:
@zshipko could you describe the rational of `Batch` here? Does it mean that `Client.Tree` always connect to the server while `Client.Batch.Tree` is maintaining a local tree?

Reply:
That's right, the `Batch` API allows you to manipulate a local tree and send it all at once since the Irmin `Store` API doesn't give much control over when data will be transferred over the network.

Reply:
Do you mind adding a bit of documentation about this somewhere? Maybe in the corresponding mli?

Reply:
I am in the process of adding some more comments, I will push something up this week